wsd/COOLWSD.hpp

```diff
@@ -198,13 +198,13 @@ class WSProcess
 
 protected:
     std::shared_ptr<WebSocketHandler> getWSHandler() const { return _ws; }
-    std::shared_ptr<StreamSocket> getSocket() const { return _socket; };
+    std::shared_ptr<StreamSocket> getSocket() const { return _socket.lock(); };
 
 private:
     std::string _name;
     std::atomic<pid_t> _pid; //< The process-id, which can be access from different threads.
-    std::shared_ptr<WebSocketHandler> _ws;
-    std::shared_ptr<StreamSocket> _socket;
+    std::shared_ptr<WebSocketHandler> _ws;  // FIXME: should be weak ? ...
```

Comment:
`WebSocketHandler` is created in `ChildProcess`'s ctor. There are no other references to it than this.

Weak works for `_socket` because it's referenced in the `SocketPoll`. So weak is good there.